src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilder.cs

```diff
@@ -40,10 +41,10 @@ public HubConnection Build()
             // The service provider is disposed by the HubConnection
             var serviceProvider = Services.BuildServiceProvider();
 
-            var connectionFactory = serviceProvider.GetService<Func<IConnection>>();
+            var connectionFactory = serviceProvider.GetService<IConnectionFactory>();
             if (connectionFactory == null)
             {
-                throw new InvalidOperationException("Cannot create HubConnection instance. A connection was not configured.");
+                throw new InvalidOperationException($"Cannot create HubConnection instance. An {nameof(IConnectionFactory)} was not configured.");
```

Comment:
If we're using nameof, also nameof(HubConnection)?

Reply:
Sounds good